File: Data Structures & Algorithms/max-water-container/submission-1.java
class Solution {
    public int maxArea(int[] heights) {

        int left =0;
        int right= heights.length-1;

        int max_cap=0;

        while(left<right){

            int len = right-left;

            if(heights[left]>heights[right]){
                if(max_cap<len*heights[right]){
                    max_cap=len*heights[right];
                }
                right--;
            }else{
                if(max_cap<len*heights[left]){
                    max_cap=len*heights[left];
                }
                left++;
            }            

        }

        return max_cap;
        
    }
}

File: Data Structures & Algorithms/house-robber-ii/submission-1.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        if len(nums) == 1:
            return nums[0]

        left = self.robber(nums[:len(nums)-1])
        right = self.robber(nums[1:])

        return max(left,right)
        

        
        
        
    def robber(self,nums):
        rob1 ,rob2 = 0,0

        for n in nums:
            temp = max(rob2,rob1+n)
            rob1 = rob2
            rob2= temp
        return rob2
        